### PR TITLE
feat: lu shell api

### PR DESCRIPTION
**Pull request by zhixzhan**
## Description

update latest BF-LU parser, add section operate api in composer shell.

1. Provide LU intent section CRUD API in shell.
2. Refactor LuFile interface and related luIndexer / server api.

## Task Item

refs #867 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code refactor (non-breaking change which improve code quality, clean up, add tests, etc)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Doc update (document update)

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have functionally tested my change

## Screenshots

Please include screenshots or gifs if your PR include UX changes.

### CODE REVIEW

**Comment by boydc2014** [comment]
@liweitian  maybe you can check, can the create new dialog wizard use this?

**Comment by zhixzhan** [comment]
And @alanlong9278, please take a look, make sure it satisfies the usage scenario.

File: Composer/packages/client/package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "@bfc/extensions": "*",
     "@bfc/indexers": "*",
     "@bfc/shared": "*",
+    "@bfcomposer/bf-lu": "^1.1.5",
```
Comment (boydc2014) [inline Composer/packages/client/package.json]: One question to @lei9444 , is this the latest version of bf-lu and is this the version which lubuild used? we need to make sure this version is consistent

Comment (lei9444) [reply Composer/packages/client/package.json]: lubuild use 1.1.2 now

**Comment by liweitian** [comment]
> @liweitian maybe you can check, can the create new dialog wizard use this?
Maybe you mean create new trigger wizard?
Talked with @zhixzhan on some code issue.
Tested on my side. An error happened when updating intent on an empty lufile. It should be a bug in @bfcomposer/bf-lu/lib/parser package.

File: Composer/packages/client/package.json (hunk 1 above)
Comment (zhixzhan) [reply Composer/packages/client/package.json]: bf-lu@1.1.2+ is some bug fix or features update, base on my feedback to Fei.

**Comment by zhixzhan** [comment]
> > @liweitian maybe you can check, can the create new dialog wizard use this?
> > @boydc2014 Maybe you mean create new trigger wizard?
> > Talked with @zhixzhan on some code issue.
> > Tested on my side. An error happened when updating intent on an empty lufile. It should be a bug in @bfcomposer/bf-lu/lib/parser package.

Talked to Fei, he will update it.

File: Composer/packages/client/package.json (hunk 1 above)
Comment (zhixzhan) [reply Composer/packages/client/package.json]: The latest version is 1.1.6 now

File: Composer/packages/client/package.json [hunk 2]
```diff
@@ -20,6 +20,7 @@
     "@bfc/extensions": "*",
     "@bfc/indexers": "*",
     "@bfc/shared": "*",
+    "@bfcomposer/bf-lu": "^1.1.6",
```
Comment (boydc2014) [inline Composer/packages/client/package.json]: Is this version contains the patch that make section id stable?

Comment (zhixzhan) [reply Composer/packages/client/package.json]: yes

File: Composer/packages/client/package.json (hunk 1 above)
Comment (boydc2014) [reply Composer/packages/client/package.json]: So 1.1.2 is OK for lubuild, right? get it

File: Composer/packages/client/src/ShellApi.ts [hunk 3]
```diff
@@ -301,6 +344,9 @@ export const ShellApi: React.FC = () => {
     apiClient.registerApi('removeLgTemplate', removeLgTemplateHandler);
     apiClient.registerApi('removeLgTemplates', removeLgTemplatesHandler);
     apiClient.registerApi('getLgTemplates', ({ id }, event) => getLgTemplates({ id }, event));
+    apiClient.registerApi('addLuIntent', addLuIntentHandler);
```
Comment (boydc2014) [inline Composer/packages/client/src/ShellApi.ts]: where is get api?

Comment (boydc2014) [reply Composer/packages/client/src/ShellApi.ts]: my expectation is through this set of api, form can update a whole lu intent, and a sub-section of lu intent

Comment (zhixzhan) [reply Composer/packages/client/src/ShellApi.ts]: > where is get api?

`luFiles` is accessible in `getState()` api, for current usage scenario seems no need to have another Fine-gained `get` api .

Comment (zhixzhan) [reply Composer/packages/client/src/ShellApi.ts]: > my expectation is through this set of api, form can update a whole lu intent, and a sub-section of lu intent



> my expectation is through this set of api, form can update a whole lu intent, and a sub-section of lu intent

get it, will make some change

**Pull request merged**